test/retrievers/test_bm25_retriever.py

```diff
@@ -0,0 +1,70 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from camel.retrievers import BM25Retriever
+
+
+def test_bm25retriever_initialization():
+    retriever = BM25Retriever()
+    assert retriever.bm25 is None
+    assert retriever.content_input_path == ""
+    assert retriever.chunks == []
+
+
+@patch('camel.loaders.UnstructuredIO')
+def test_process(mock_unstructured_modules):
+    mock_unstructured_modules.return_value.parse_file_or_url.return_value = [
+        'Your parsed content'
+    ]
+    mock_unstructured_modules.return_value.chunk_elements.return_value = [
+        'Chunk 1', 'Chunk 2'
+    ]
+
+    retriever = BM25Retriever()
+    retriever.process('https://www.camel-ai.org/')
+
+    assert retriever.content_input_path == 'https://www.camel-ai.org/'
+    assert len(retriever.chunks) == 4
+
+
+@patch('camel.retrievers.BM25Retriever')
+def test_query(mock_bm25):
+    # Setup the mock BM25 instance
+    mock_bm25_instance = mock_bm25.return_value
+    mock_bm25_instance.get_scores.return_value = [0.8, 0.5]  # Mock scores
+
+    # Create mock chunks with `metadata` attribute
+    mock_chunk1 = MagicMock(text='Chunk 1 text')
+    mock_chunk1.metadata.to_dict.return_value = {'id': 'chunk1'}
+    mock_chunk2 = MagicMock(text='Chunk 2 text')
+    mock_chunk2.metadata.to_dict.return_value = {'id': 'chunk2'}
+
+    retriever = BM25Retriever()
+    retriever.bm25 = mock_bm25_instance
+    retriever.content_input_path = 'dummy_path'
+    retriever.chunks = [mock_chunk1, mock_chunk2]
+
+    results = retriever.query('query', top_k=2)
+    assert results[0]['similarity score'] == 0.8
+    assert 'chunk1' in results[0]['metadata']['id']
+
```

Comment:
Should better consider about the top_k, not only top_1.

Reply:
added